posthog/models/feature_flag.py

```diff
@@ -75,6 +89,15 @@ def __init__(self, distinct_id: str, feature_flag: FeatureFlag):
     def is_match(self):
         return any(self.is_group_match(group, index) for index, group in enumerate(self.feature_flag.groups))
 
+    def get_matching_variant(self) -> Optional[str]:
+        if not self.is_match():
+            return None
+
+        for variant in self.variant_lookup_table:
+            if self._hash_md5 >= variant["value_min"] and self._hash_md5 < variant["value_max"]:
+                return variant["key"]
+        return self.feature_flag.fallback_variant_key
```

Comment:
In which type of scenario would we expect fallback to be hit?

Could you maybe walk me through an example flag setup where we'd expect fallback to be the return instead one of the variants?

As a followup, in which scenarios would we opt for a fallback as opposed to an additional variant? Given we are requiring the variant allocations to sum to 100%

Reply:
I think this would be most helpful if/when we roll out support to do further filtering for each variant. The bucketing might not always be based on the hash evaluation and rollout percentage.

Basically this sets the stage for the last example in the "candidates for future work" [here](https://github.com/PostHog/product-internal/issues/138#issuecomment-900452577).

Imagine you want to roll out a header image based on user properties.
1. `city = San Francisco` -> return `Golden Gate`
2. `city = New York` -> return `Brooklyn Bridge`
3. fallback -> return `generic stock photo`

If we didn't have a fallback option, the third variant would have to be `city != San Francisco and city != New York ... ` and so on

Reply:
Thanks for summarizing that for me, a few more questions.

- The use case of city = San Francisco, city = New York returning different variants – we're not building that for this MVP, (i.e. each flag and all its variants will be working with the same audience), is that right?

I'm wondering if we should defer building this out and adding code for fallbacks until we are ready to tackle that part of the problem. 

At minimum, I don't think we should have any reference to fallbacks on the front-end of this flow, since I think it is simpler for a user to understand all the variants need to sum to 100.

Also circling back to launchdarkly – their use of what I think we're calling fallback_variant is a little bit different.

In their setup, this refers to what to do when "targeting is off"–

In PostHog terms, that would mean someone has hit the green switch on the flag to disable it – so I feel like the area in your code where we'd be throwing people into the fallback (if we are trying to use their semantics) would be higher up, in the initial code where you are grabbing feature flags and checking for the top-level match. 

<img width="1008" alt="Screen Shot 2021-08-19 at 7 01 35 PM" src="https://user-images.githubusercontent.com/5965891/130167670-520be1bb-cd07-40c0-a511-ffd0d0877b11.png">

Reply:
Ok, can definitely strip this out-- do you think it'd be more intuitive to return the fallback value when targeting is off, or return None/undefined and we'll consider how to add fallback values later?

Reply:
So my gut is telling me the best thing here is to try and stay as consistent with how we treat it today – right now, when a flag is off it's omitted from the result, so maybe we should do that?

Then in the future when we want to add more support here we can think about how we want to treat fallbacks in the multivariate and non-multivariate case